framework/physics/FieldFunction/ff_gridbased_00_constrdestr.cc

```diff
@@ -24,6 +24,10 @@ chi::InputParameters FieldFunctionGridBased::GetInputParameters()
   params.AddOptionalParameter(
     "initial_value", 0.0, "The initial value to assign to the field function");
 
+  using namespace chi_data_types;
+  params.ConstrainParameterRange(
+    "sdm_type", AllowableRangeList::New({"FiniteVolume", "PWLC", "PWLD"}));
```

Comment:
Should we be consistent and use "FV" for finite volume?

Reply:
I've spent some thought on this. Consider this: I thought about likely new additions to the spatial discretization family and Lagrange came to mind as the likely next candidate. In fact we could have a hybrid Lagrange but PWL sprinkled in if a polygon or polyhedron is in there. With that in regard I singled out the more wordy SDMs and shortened them. So Instead of having `FV, PWLC, PWLD, LG?` we would have `FiniteVolume, Lagrange, PWLC, PWLD, DiscontinuousLagrange, Monomial, Raviart-Thomas, etc.` 

Would be a shame to have `PieceWiseLinear` and `PieceWiseLinearDiscontinuous` in there for the sake of consistency, however, I could live with it. Would you like me then to write them all out?

Reply:
It's safe for me to say that I don't think the all-abbreviated case would be good.
Which option do you prefer here:
```lua
tbulk = chi_physics.FieldFunctionGridBased.Create({
  name = "T_bulk",
  --sdm_type = "FiniteVolume",
  --sdm_type = "PiecewiseLinear",
  --sdm_type = "PiecewiseLinearDiscontinuous",
  --sdm_type = "Lagrange",
  --sdm_type = "LagrangeDiscontinuous"
  --sdm_type = "Monomial"
  --sdm_type = "RaviartThomas"
  initial_value = 50.0
})

tbulk = chi_physics.FieldFunctionGridBased.Create({
  name = "T_bulk",
  --sdm_type = "FiniteVolume",
  --sdm_type = "PWLC",
  --sdm_type = "PWLD",
  --sdm_type = "Lagrange",
  --sdm_type = "LagrangeD"
  --sdm_type = "Monomial"
  --sdm_type = "RaviartThomas"
  initial_value = 50.0
})
```

Reply:
Regardless, the user would have to consult documentation. I am just rooting for not spelling out Discontinuououuoss every time.

Reply:
My thought is that we should abbreviate for now, and then revisit when we have those other options available. I would lean towards "standard" methods being abbreviated (i.e. methods often referred to with an abbreviation) and non-standard methods being spelt out.

Reply:
Ok. I will then make it all abbreviated. Just bare in mind that Lagrange is pretty close to being reality so then I would prefer `Langrange` and `LagrangeD`

Reply:
Done